sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/components/EmailGenerator.groovy

```diff
@@ -1,48 +1,65 @@
 package life.qbic.samplenotificator.components
 
-import life.qbic.business.subscription.Subscriber
+import life.qbic.business.notification.create.NotificationContent
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
 
 import java.util.concurrent.TimeUnit
 
+
 /**
  * Class responsible for generating an sending an email from the provided notification and subscriber information
  *
  * EmailGenerator converts the notification created in the CreateNotification Use case into an emailBody
  * The contactInformation such as e.g. the emailAddress of the recipient is retrieved from the associated provided Subscriber
  * The email is then send via a commandline call to the mailutils tool(@link <a href=https://mailutils.org/>mailutils</a> via a ProcessBuilder
  *
- * @since: 1.0.0
- *
+ * @since: 1.0.0*
  */
 class EmailGenerator {
 
     private String subject = "Project Update"
-    Map<Subscriber, String> notificationPerSubscriber
+    List<NotificationContent> notificationContentList
+    private InputStream EMAIL_HTML_TEMPLATE_STREAM
+    private EmailHTMLTemplate emailHTMLTemplate
+    private List<Document> filledTemplates = []
 
-    EmailGenerator(Map<Subscriber, String> notificationPerSubscriber) {
-        this.notificationPerSubscriber = notificationPerSubscriber
+    EmailGenerator(List<NotificationContent> notificationContentList) {
+        this.notificationContentList = notificationContentList
```

Comment:
I would expect it to have a fillTemplate(List notifications) method rather than do it like this

Reply:
Oh sure that's a good Idea!